python/src/trezorlib/cli/cardano.py

```diff
@@ -59,14 +60,46 @@ def sign_tx(client, file, protocol_magic):
 
 
 @cli.command()
-@click.option("-n", "--address", required=True, help=PATH_HELP)
+@click.option("-n", "--path-str", required=True, help=PATH_HELP)
 @click.option("-d", "--show-display", is_flag=True)
-@click.option("-p", "--protocol-magic", type=int, default=1)
+@click.option(
+    "-t", "--address-type", type=int, default=messages.CardanoAddressType.BASE
```

Comment:
please use `ChoiceType`, see `btc.py` for example

if you want users to be able to supply ints as well, just add the same value to the dict multiple times

Reply:
Done [here](https://github.com/trezor/trezor-firmware/pull/1112/commits/8c4a5c8a733d9d991405e5ee1c4f9d9bfe9895bf). I wasn't aware of this and I think it's better than ints.